projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -225,12 +238,69 @@ export class DatagridComponent<R> implements OnInit {
     ngOnInit(): void {
         this.isLoading = true;
         this.gridRefresh.emit({});
+        this.clearSelectionInformation();
+    }
+
+    private updateSelectedItems(): void {
```

Comment:
Can we write two tests?

* Selection is maintained after a refresh if record is still available
* Selection is cleared after a refresh is record is no longer available

Reply:
Had a test to test your second bullet point. Added a test to explicitly test the first.